java/arcs/core/data/expression/Builders.kt

```diff
@@ -189,7 +189,9 @@ open class MapScope<V>(
     override val scopeName: String,
     val map: MutableMap<String, V>
 ) : Scope {
-    override fun <V> lookup(param: String): V = map[param] as V
+    override fun <V> lookup(param: String): V = if (map.containsKey(param)) {
+        map[param] as V
+    } else throw IllegalArgumentException("Field '$param' not found")
```

Comment:
nit: add "field '$param' not found on scope '$scopeName'"

Reply:
You could also use [MutableMap.getValue](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/get-value.html) if you prefer.

Reply:
Thanks Gogul! I will go with Ray's suggestion in order to customize the error message.